Controller/RoleController.php

```diff
@@ -0,0 +1,194 @@
+use eZ\Publish\API\Repository\RoleService;
+use eZ\Publish\Core\MVC\Symfony\Security\Authorization\Attribute;
+use eZ\Publish\Core\Repository\Values\User\RoleCreateStruct;
+use EzSystems\RepositoryForms\Data\Mapper\RoleMapper;
+use EzSystems\RepositoryForms\Form\ActionDispatcher\ActionDispatcherInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class RoleController extends Controller
+{
+    /**
+     * @var \eZ\Publish\API\Repository\RoleService
+     */
+    protected $roleService;
+
+    /**
+     * @var ActionDispatcherInterface
+     */
+    private $actionDispatcher;
+
+    public function __construct(
+        RoleService $roleService,
+        ActionDispatcherInterface $actionDispatcher
+    ) {
+        $this->roleService = $roleService;
+        $this->actionDispatcher = $actionDispatcher;
+    }
+
+    /**
+     * Renders the role list.
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function listRolesAction()
+    {
+        return $this->render('eZPlatformUIBundle:Role:list_roles.html.twig', [
+            'roles' => $this->roleService->loadRoles(),
+            'can_edit' => $this->isGranted(new Attribute('role', 'update')),
+            'can_create' => $this->isGranted(new Attribute('role', 'create')),
+            'can_delete' => $this->isGranted(new Attribute('role', 'delete')),
+        ]);
+    }
+
+    /**
+     * Renders a role.
+     *
+     * @param int $roleId Role ID
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function viewRoleAction($roleId)
+    {
+        $role = $this->roleService->loadRole($roleId);
+        $roleAssignments = $this->roleService->getRoleAssignments($role);
+
+        return $this->render('eZPlatformUIBundle:Role:view_role.html.twig', [
+            'role' => $role,
+            'role_assignments' => $roleAssignments,
+            'can_edit' => $this->isGranted(new Attribute('role', 'update')),
+            'can_delete' => $this->isGranted(new Attribute('role', 'delete')),
+        ]);
+    }
+
+    /**
+     * Creates a role.
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function createRoleAction()
+    {
+        try {
+            $roleCreateStruct = new RoleCreateStruct([
+                'identifier' => '__new__' . md5(microtime(true)),
+            ]);
+            $role = $this->roleService->createRole($roleCreateStruct);
+        } catch (UnauthorizedException $e) {
+            return $this->forward('eZPlatformUIBundle:Pjax:accessDenied');
+        } catch (InvalidArgumentException $e) {
+            return $this->forward('eZPlatformUIBundle:Pjax:accessDenied'); // TODO: is this correct? Name is already existing, or limitation errors
+        } catch (LimitationValidationException $e) {
+            return $this->forward('eZPlatformUIBundle:Pjax:accessDenied'); // TODO: is this correct? Invalid policy limitation.
```

Comment:
as its name suggests, `eZPlatformUIBundle:Pjax:accessDenied` is supposed to be called when the user does not have access to something. So calling it for other cases is not correct. For the others Exception, most of the time, you should add a notification message like https://github.com/ezsystems/PlatformUIBundle/blob/master/Controller/SectionController.php#L287